### ISSUE TITLE
mypy does not recognize sys.platform with ternary operator

### ISSUE THREAD

**Issue by amatyukhin0** [python/mypy#13003]
**Bug Report**

mypy can recognize `sys.platform` in `if cond: ... / else: ...` conditions but fails to do it if ternary operator (`a if cond else b`) is used.

**To Reproduce**

Consider this piece of code, assuming we are on Linux:

```
import signal
import sys

signal_type = signal.SIGBREAK if sys.platform == 'win32' else signal.SIGTERM
```

**Expected Behavior**

mypy should not raise any error.

**Actual Behavior**

mypy fails with:

```
main.py:4: error: Module has no attribute "SIGBREAK"
Found 1 error in 1 file (checked 1 source file)
```

**Environment**

- Mypy version used: 0.961
- Python version used: 3.10
- Operating system: Linux

**Comment by AlexWaygood** [comment]
I think this is better classified as a feature rather than a bug, since [PEP 484](https://peps.python.org/pep-0484/#version-and-platform-checking) states:

> Type checkers are expected to understand _simple_ version and platform checks

(Emphasis mine on "simple".)
